source/code/providers/support/mycnf-scanner.h

```diff
@@ -0,0 +1,155 @@
+/*
+ * --------------------------------- START OF LICENSE ----------------------------
+ *
+ * MySQL cimprov ver. 1.0
+ *
+ * Copyright (c) Microsoft Corporation
+ *
+ * All rights reserved. 
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the ""Software""), to deal
+ * in the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+ * of the Software, and to permit persons to whom the Software is furnished to do
+ * so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ * ---------------------------------- END OF LICENSE -----------------------------
+*/
+ /**
+        \file        mycnf-scanner.h
+
+        \brief       Parses MySQL configuration files to extract authentication info.
+
+        \date        06-09-16
+*/
+/*----------------------------------------------------------------------------*/
+
+#ifndef MYCNF_SCANNER_H
+#define MYCNF_SCANNER_H
+#include <iostream>
+#include <string>
+#include <vector>
+#include <set>
+#include <map>
+
+#include <scxcorelib/scxdirectoryinfo.h>
+#include <scxcorelib/scxexception.h>
+#include <scxcorelib/scxfile.h>
+#include <scxcorelib/scxfilepath.h>
+#include <scxcorelib/scxfilesystem.h>
+#include <scxcorelib/scxhandle.h>
+#include <scxcorelib/scxlog.h>
+#include <scxcorelib/stringaid.h>
+
+#include "sqlauth.h"
```

Comment:
I will look into these, thanks. I think one/two of ```scxfile.h```, ```scxfilepath.h``` and ```scxfilesystem.h``` are unnecessary. 
Others are needed.